pkg/model/config/config.go

```diff
@@ -78,39 +81,105 @@ func (c Config) IsV3() bool {
+}
+
+func (c Config) errorForField(fieldName string) error {
+	return unsupportedFieldError{fieldName: fieldName, version: c.Version}
+}
+
+// validate checks if a Config is valid or it has some fields that shouldn't be present for that version
+func (c Config) validate() error {
+	if c.IsV2() {
+		if c.ProjectName != "" {
+			return c.errorForField("projectName")
+		}
+		for _, r := range c.Resources {
+			if r.Domain != "" {
+				return c.errorForField("resources[].domain")
+			}
+			if r.Plural != "" {
+				return c.errorForField("resources[].plural")
+			}
+			if r.Path != "" {
+				return c.errorForField("resources[].plural")
+			}
+			if r.API != nil {
+				return c.errorForField("resources[].api")
+			}
+			if r.Controller {
+				return c.errorForField("resources[].controller")
+			}
+			if r.Webhooks != nil {
+				return c.errorForField("resources[].webhooks")
+			}
+		}
+		if c.ComponentConfig {
+			return c.errorForField("componentConfig")
+		}
+		if c.Layout != "" {
+			return c.errorForField("layout")
+		}
+		if len(c.Plugins) != 0 {
+			return c.errorForField("plugins")
+		}
+	}
+
+	return nil
+}
+
+// GetResource returns the requested resource if it is already tracked
+func (c Config) GetResource(target resource.GVK) *resource.Resource {
+	// Return the target resource if it is found in the tracked resources
 	for _, r := range c.Resources {
-		if r.isEqualTo(target) {
-			return true
+		if r.GVK().IsEqualTo(target) {
+			return r
 		}
 	}
 
-	// Return false otherwise
-	return false
+	// Return nil otherwise
+	return nil
 }
 
-// UpdateResources either adds gvk to the tracked set or, if the resource already exists,
-// updates the the equivalent resource in the set.
-func (c *Config) UpdateResources(gvk GVK) {
+// UpdateResources adds the resource to the tracked ones or updates it as needed
+func (c *Config) UpdateResources(res *resource.Resource) (*resource.Resource, error) {
+	// Create a copy of the resource to ensure we do not accidentally modify it
+	resCopy := *res
+
+	// V2 only requires group, version and kind fields
+	if c.IsV2() {
+		resCopy = resource.Resource{
+			Group:   resCopy.Group,
+			Version: resCopy.Version,
+			Kind:    resCopy.Kind,
+		}
+	}
+
+	gvk := resCopy.GVK()
 	// If the resource already exists, update it.
 	for i, r := range c.Resources {
-		if r.isEqualTo(gvk) {
-			c.Resources[i].merge(gvk)
-			return
+		if r.GVK().IsEqualTo(gvk) {
+			err := c.Resources[i].Update(&resCopy)
+			return c.Resources[i], err
 		}
 	}
 
 	// The resource does not exist, append the resource to the tracked ones.
-	c.Resources = append(c.Resources, gvk)
+	c.Resources = append(c.Resources, &resCopy)
+	return res, nil
 }
 
 // HasGroup returns true if group is already tracked
 func (c Config) HasGroup(group string) bool {
 	// Return true if the target group is found in the tracked resources
 	for _, r := range c.Resources {
-		if strings.EqualFold(group, r.Group) {
+		if strings.EqualFold(group, r.QualifiedGroup()) {
```

Comment:
IMO we cannot do it. It changes the default behaviour. We do not use the qualified group with the domain to do this check and IMO we should not introduce this new logic/rule her. it has a big chance to impacts the v2 behaviours and  introduces a ranking change besides not be part of the scope of this pr.

Reply:
It doesn't because for v2 project configuration, domain is not stored in the `Resources []*Resource` field, so the `QualifiedGroup` is the same as the group.

Reply:
This is susceptible to errors and changes behavioural. In addition, the HasGrupo function should not use the domain. As the name says it should only and check if the group exists or not. (please do not close this comment as solved) 

c/c @estroz

Reply:
It sounds like @Adirio is explicitly saying that behavior _isn't_ changing by using `Resource.QualifiedGroup()` here, because v2 configs consider a qualified group to be just `Resource.Group` in the context of that resource.

Reply:
The func is called HasGroup which ` returns true if the group is already tracked`. However, its code is to check the Group + Domain. 

Note that in a follow a contributor can for some reason starts to set the domain then, it will starts to fails. also, I do not think that we should change the logic and how the business rules for v2 unless it is really required to achieve to solve a  problem which is not this case. Otherwise, we are unnecessarily assuming a risk.

Reply:
We don't believe that the domain is empty, we enforce it. The domain in v2 project files is enforced to be empty, and therefore the qualified group in this context is only the group, which matches with the `GVK` generated by v2 options.